### PR TITLE
W3 signer token

### PR DESCRIPTION
**Pull request by ice-breaker-tg**
Extends options for the signer service and the SIGV4 to allow passing the security token generated when a temp IAM is used instead of a user based one.

### CODE REVIEW

**Comment by Gozala** [comment]
@hugomrdias I don't know why this is still failing, it should be fixed by #67 and it is on my device. I suspect there is some caching happening and new changes in target branch aren't picked up on reruns.

**Comment by Gozala** [comment]
@ice-breaker-tg do you mind doing `git fetch && git merge origin/main && git push` which I believe should fix the failing test. Since pr is from your fork, I'm unable to add any commits here.

**Comment by ice-breaker-tg** [comment]
> @ice-breaker-tg do you mind doing `git fetch && git merge origin/main && git push` which I believe should fix the failing test. Since pr is from your fork, I'm unable to add any commits here.

Did this, ran tests locally and they all passed so hopefully should be good.
Let me know if there is anything else.

File: w3/sigv4/src/index.js [hunk 1]
```diff
@@ -55,6 +56,9 @@ export class SigV4 {
       this.accessKeyId + '/' + this.credentialString
     )
     params.set('X-Amz-SignedHeaders', this.signedHeaders)
+    if (this.securityToken) {
+      params.set('x-amz-security-token', this.securityToken)
```
Comment (hugomrdias) [inline w3/sigv4/src/index.js:60]:
```suggestion
      params.set('X-Amz-Security-Token', this.securityToken)
```

**Review by hugomrdias — CHANGES_REQUESTED** [review]
Can you please change the param name to `sessionToken` to keep it the same as the s3 sdk ?

**Review by hugomrdias — APPROVED** [review]

**Pull request merged**